pkg/server/server.go

```diff
@@ -352,7 +409,7 @@ func NewServerWithOpts(opts ...OpenFGAServiceV1Option) (*Server, error) {
 			graph.WithLogger(s.logger),
 			graph.WithCacheTTL(s.checkQueryCacheTTL),
 		)
-		s.cachedCheckResolverCloser = cachedCheckResolver.Close
+		s.cachedCheckResolver = cachedCheckResolver
 
 		cachedCheckResolver.SetDelegate(localChecker)
```

Comment:
You're missing a setup of the `dispatchThrottlingCheckResolver` in here.

There should be a:
```go
if dispatchThrottlingChecResolver != nil {
    disptachThrottlingCheckResolver.SetDelegate(cachedCheckResolver)
}
```
in this block of code somewhere.

The order is this `CycleDetectionCheckResolver` --> (optional) `DispatchThrottlingCheckResolver` --> (optional) `CachedCheckResolver` -->` LocalChecker` --> `CycleDetectionCheckResolver`

Reply:
test added